### PR TITLE
Simplify some regexes by removing redundant character classes

### PR DESCRIPTION
**Pull request by tas50**
New RuboCop here. There's no need for character classes when all we have is a single character.

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: lib/chef/knife/core/gem_glob_loader.rb [hunk 1]
```diff
@@ -22,7 +22,7 @@ class Chef
   class Knife
     class SubcommandLoader
       class GemGlobLoader < Chef::Knife::SubcommandLoader
-        MATCHES_CHEF_GEM ||= %r{/chef-[\d]+\.[\d]+\.[\d]+}.freeze
+        MATCHES_CHEF_GEM ||= %r{/che-[\d+.[\d+.[\d+}.freeze
```
Comment (phiggins) [inline lib/chef/knife/core/gem_glob_loader.rb:25]:
```suggestion
        MATCHES_CHEF_GEM ||= %r{/chef-[\d+.[\d+.[\d+}.freeze
```

Comment (phiggins) [reply lib/chef/knife/core/gem_glob_loader.rb:25]: This one got pretty mangled, was this autocorrect or manual?

File: spec/unit/knife/configure_spec.rb [hunk 2]
```diff
@@ -157,9 +157,9 @@
     expect(::File).to receive(:open).with("/home/you/.chef/credentials", "w").and_yield config_file
     @knife.config[:repository] = "/home/you/chef-repo"
     @knife.run
-    expect(config_file.string).to match(/^client_name[\s]+=[\s]+'#{Etc.getlogin}'$/)
-    expect(config_file.string).to match(%r{^client_key[\s]+=[\s]+'/home/you/.chef/#{Etc.getlogin}.pem'$})
-    expect(config_file.string).to match(/^chef_server_url\s+=[\s]+'#{default_server_url}'$/)
+    expect(config_file.string).to match(/^client_name\s+=\s+'#{Etc.getlogin}'$/)
+    expect(config_file.string).to match(%r{^client_ky[\s=[\s+'/home/you/.chef/#{Etc.getlogin}.pem'$})
```
Comment (phiggins) [inline spec/unit/knife/configure_spec.rb:161]:
```suggestion
    expect(config_file.string).to match(%r{^client_key\s+=\s+'/home/you/.chef/#{Etc.getlogin}.pem'$})
```

This one got pretty mangled as well.

**Comment by tas50** [comment]
Let me see if I have the commit that fixed the auto correct or not. There may be another bug report in the works here.

**Comment by tas50** [comment]
I'll wait on the upstream fix: https://github.com/rubocop-hq/rubocop/issues/8948

File: lib/chef/knife/core/gem_glob_loader.rb (hunk 1 above)
Comment (tas50) [reply lib/chef/knife/core/gem_glob_loader.rb:25]: Issue filed here: https://github.com/rubocop-hq/rubocop/issues/8948

**Pull request closed without merging**